airbyte-integrations/connectors/source-google-pagespeed-insights/source_google_pagespeed_insights/google_pagespeed_insights.yaml

```diff
@@ -0,0 +1,60 @@
+version: "0.1.0"
+definitions:
+  selector:
+    extractor:
+      field_pointer: []
+  url_stream_slicer:
+    type: "ListStreamSlicer"
+    slice_values: "{{ config['urls'] }}"
+    cursor_field: url
+    request_option:
+      field_name: "url"
+      inject_into: "request_parameter"
+  strategy_stream_slicer:
+    type: "ListStreamSlicer"
+    slice_values: "{{ config['strategies'] }}"
+    cursor_field: strategy
+    request_option:
+      field_name: "strategy"
+      inject_into: "request_parameter"
+  category_stream_slicer:
```

Comment:
Categories do not need different API call, as a single API call accepts multiple categories.

So I don't think they should be included in the slicers :) 

(Also, to be confirmed, but I don't think it is **currently** useful here, as I think it only affect the Lighthouse results, which we don't save here!)

Reply:
Yeah the problem is the lowcode cdk I think: you can't define multiple constant request parameters with the same key, right?
Any suggestion how one can circumvent this without the stream slicer workaround?

Reply:
> (Also, to be confirmed, but I don't think it is currently useful here, as I think it only affect the Lighthouse results, which we don't save here!) 

Ok good point - actually the schema is missing the lighthouse report sections. I removed them in the beginning because it's quite hard to find a schema for alle web-pages (as the lighthouse report section contains a lot of objects based on the structure of the website).
Nevertheless we are reading the lighthouse results and therefore we are also storing them, right? But as the schema is missing some fields, we can't normalize the results correctly? Or what's actually the impact, if there are fields in the streams which are not defined in the schema?

What would you suggest here? Could we imply add the "lighthouseResult" property as an "object" - type to the schema for now? Alternatively I can look through all the properties and manually select the ones which are "universal" - but this will most probably take some days.

Reply:
> Do you have any suggestions for how one can add multiple request parameters with the same key? This would be required if we go without this streamslicer workaround.

I'm not sure if this is a hack or if its intended, but this seems to be working:
```yaml
      request_parameters:
        key: "{{ config['api_key'] }}"
        category: "{{ config['categories'] }}"
```

(if we use this, we should make sure a test is covering it, so that we're sure it keeps working in the future)

>Nevertheless we are reading the lighthouse results and therefore we are also storing them, right?
For most destinations that go through normalisation, yes, however I'm not sure there is an easy way to "renormalize" the entire dataset in the future.

I'm using the bigquery denormalized destination a lot, and in my case the data would be lost :)

I think it's acceptable to not add the lighthouse results at all, and leave it for a further improvement.

I would not be against adding a simplified version of it either, with a few objects properties here and there when its undocumented. I'll let you decide!